keel-core/src/main/kotlin/com/netflix/spinnaker/keel/persistence/memory/InMemoryArtifactRepository.kt

```diff
@@ -138,6 +146,36 @@ class InMemoryArtifactRepository : ArtifactRepository {
     }
   }
 
+  override fun versionsByEnvironment(deliveryConfig: DeliveryConfig): List<EnvironmentArtifactsSummary> =
+    deliveryConfig
```

Comment:
I find this hard to parse, with the super nested structure. Any way we can simplify to make it easier to understand what's going on?